src/UI/Art/SpriteArtist/OverlayArtist.java

```diff
@@ -0,0 +1,110 @@
+package UI.Art.SpriteArtist;
+
+import java.awt.AlphaComposite;
+import java.awt.Color;
+import java.awt.Composite;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import Engine.XYCoord;
+import Terrain.GameMap;
+import UI.GameOverlay;
+
+public class OverlayArtist
+{
+  private static final Color HIGHLIGHT_COLOR = new Color(255, 255, 255, 160); // white
+
+  public static final Color MOBILE_FIRE_EDGE = new Color(240,   0,   0, 255); // red
+  public static final Color SIEGE_FIRE_EDGE  = new Color(  0,   0, 160, 255); // dark blue
+  public static final Color FIRE_FILL        = new Color(  0,   0,   0,   0);
+
+  private static final int OVERLAY_EDGE_THICKNESS = 2;
+
+  static AlphaComposite buffComposite = null;
+  static long lastCompositeCreationTime = 0;
+
+  public static void drawHighlights(Graphics bigG,
+                                    GameMap gameMap,
+                                    Collection<GameOverlay> inputOverlays,
+                                    int drawX, int drawY,
+                                    int viewWidth, int viewHeight,
+                                    int tileSize)
+  {
+    if( viewWidth < 1 || viewHeight < 1 )
+      return;
+
+    BufferedImage overlayImage = SpriteLibrary.createTransparentSprite(viewWidth, viewHeight);
+    Graphics og = overlayImage.getGraphics();
+
+    // Only show stuff we're allowed to
+    Collection<GameOverlay> overlays = new ArrayList<GameOverlay>();
+    for( GameOverlay ov : inputOverlays )
+      if( null == ov.origin || !gameMap.isLocationFogged(ov.origin) )
+        overlays.add(ov);
+
+    for( int w = 0; w < gameMap.mapWidth; ++w )
+    {
+      if( (w+1)*tileSize < drawX || drawX + viewWidth < (w-2)*tileSize )
+        continue;
+      for( int h = 0; h < gameMap.mapHeight; ++h )
+      {
+        if( (h+1)*tileSize < drawY || drawY + viewHeight < (h-2)*tileSize )
+          continue;
+
+        XYCoord coord = new XYCoord(w,h);
+        for( GameOverlay ov : overlays )
+          if( ov.area.contains(coord) )
+            drawOverlayTile(og, ov, coord, w * tileSize - drawX, h * tileSize - drawY, tileSize);
+        
+        if( gameMap.isLocationValid(w, h) )
+        {
+          Terrain.Location locus = gameMap.getLocation(w, h);
+          if( locus.isHighlightSet() )
+          {
+            og.setColor(HIGHLIGHT_COLOR);
+            og.fillRect(w * tileSize - drawX, h * tileSize - drawY, tileSize, tileSize);
+          }
+        }
+      }
+    }
+
+    // Set opacity as a function of time.
+    long nowTime = System.currentTimeMillis();
+//    float overlayOpacity = (float)(0.5*Math.max(0, Math.sin(nowTime/420.)));
+    float overlayOpacity = (float)(0.15*Math.sin(nowTime/420.) + 0.7);
```

Comment:
Hey this looks familiar.

Reply:
\*gasp* I've been caught
Having variable transparency is pretty ideal for visibility to both things, but I didn't want it changing drastically, so...